aws_advanced_python_wrapper/limitless_connection_plugin.py

```diff
@@ -0,0 +1,531 @@
+        retry_count = 0
+        max_retries = WrapperProperties.MAX_RETRIES_MS.get_int(context.get_props())
+        while retry_count < max_retries:
+            retry_count += 1
+            if context.get_limitless_routers() is None or len(context.get_limitless_routers()) == 0 or not context.is_any_router_available():
+                self._synchronously_get_limitless_routers_with_retry(context)
+
+                if (context.get_limitless_routers() is None
+                        or len(context.get_limitless_routers()) == 0
+                        or not context.is_any_router_available()):
+                    logger.debug("LimitlessRouterServiceImpl.NoRoutersAvailableForRetry")
+
+                    if context.get_connection() is not None and not self._plugin_service.driver_dialect.is_closed(context.get_connection()):
+                        return
+                    else:
+                        try:
+                            context.set_connection(context.get_connect_func()())
+                            return
+                        except Exception as e:
+                            if self._is_login_exception(e):
+                                raise e
+
+                            raise AwsWrapperError(Messages.get_formatted("LimitlessRouterService.UnableToConnectNoRoutersAvailable"),
+                                                  context.get_host_info().host) from e
+
+            try:
+                selected_host_info = self._plugin_service.get_host_info_by_strategy(
+                    HostRole.WRITER, "highest_weight", context.get_limitless_routers())
+                logger.debug("LimitlessRouterServiceImpl.SelectedHostForRetry",
+                             "None" if selected_host_info is None else selected_host_info.host)
+                if selected_host_info is None:
+                    continue
+
+            except UnsupportedOperationError as e:
+                logger.error("LimitlessRouterServiceImpl.IncorrectConfiguration")
+                raise e
+            except AwsWrapperError:
+                continue
+
+            try:
+                context.set_connection(self._plugin_service.connect(selected_host_info, context.get_props(), context.get_connection_plugin()))
+                if context.get_connection() is not None:
+                    return
+
+            except Exception as e:
+                if self._is_login_exception(e):
+                    raise e
+                selected_host_info.set_availability(HostAvailability.UNAVAILABLE)
+                logger.debug("LimitlessRouterServiceImpl.FailedToConnectToHost", selected_host_info.host)
+
+        raise AwsWrapperError(Messages.get("LimitlessRouterService.MaxRetriesExceeded"))
+
+    def _synchronously_get_limitless_routers_with_retry(self, context: LimitlessConnectionContext) -> None:
+        logger.debug("LimitlessRouterServiceImpl.SynchronouslyGetLimitlessRouters")
```

Comment:
```suggestion
        logger.debug("LimitlessRouterService.SynchronouslyGetLimitlessRouters")
```

Reply:
There are a few other instances of this, might be more convenient to just do a replace all `logger.debug("LimitlessRouterServiceImpl`